packages/playwright-test/src/util.ts

```diff
@@ -195,8 +195,18 @@ export function expectTypes(receiver: any, types: string[], matcherName: string)
   }
 }
 
-export function sanitizeForFilePath(s: string) {
-  return s.replace(/[\x00-\x2C\x2E-\x2F\x3A-\x40\x5B-\x60\x7B-\x7F]+/g, '-');
+export function sanitizeForFilePath(input: string) {
+  let nonTrivialSubstitute = false;
+  let sanitized = input.replace(/[\x00-\x2C\x2E-\x2F\x3A-\x40\x5B-\x60\x7B-\x7F\x2A]+/g, substring => {
```

Comment:
I hope this list includes dashes? We want `nonTrivialSubstitute` to be true in this case.

Reply:
added a test for it.